architecture/2025-04-compute-platform-extensibility.md

```diff
@@ -0,0 +1,505 @@
+
+<!--
+### Architecture Diagram
+Provide a diagram of the system architecture, illustrating how different
+components interact with each other in the context of this proposal.
+
+Include separate high level architecture diagram and component specific diagrams, wherever appropriate.
+-->
+
+### Detailed Design
+
+#### Key Components and Relationships
+
+1. Azure Container Instances (ACI) Integration
+    * `corerp/renderers/aci` implements ACI-specific resource rendering logic
+    * `corerp/renderers/aci/gateway` provides gateway functionality for ACI
+    * `corerp/renderers/aci/manualscale` handles scaling for ACI containers
+1. Renderer Multiplexer
+    * `corerp/renderers/mux` routes requests to appropriate renderers based on compute environment
+    * Selects between Kubernetes and ACI renderers based on environment configuration
+
+This architecture supports multi-cloud deployments with a unified programming model, where applications can be defined once and deployed to either Kubernetes or Azure Container Instances environments based on the environment configuration.
+
+#### Understanding the Radius/ACI Architecture Relationships
+
+#### Renderer Architecture
+
+The rendering system is a key part of the architecture with several important relationships:
+
+1. Multiplexed Rendering
+    * The `mux.Renderer` acts as a router that directs rendering requests to the appropriate platform-specific renderer
+    * It examines the `EnvironmentComputeKind` (Kubernetes or ACI) from the environment configuration and selects the correct renderer chain
+1. Layered Rendering
+    * Renderers are organized in a layered pattern using composition:
+   KubernetesCompute: kubernetesmetadata → manualscale → daprextension → container
+   ACICompute: aci_manualscale → aci
+    * Each layer handles a specific aspect of rendering, with inner renderers focusing on core functionality and outer renderers adding specialized features
+
+#### Key Type Relationships in the Multiplexer Pattern
+
+1. Core Interface Hierarchy
+    * `renderers.Renderer` is the central interface that defines the rendering contract
+    * All concrete renderers implement this interface, providing a consistent API
+1. Multiplexer Implementation
+    * `mux.Renderer` implements the `renderers.Renderer` interface but delegates to inner renderers
+    * It maintains a map of inner renderers keyed by `EnvironmentComputeKind` (Kubernetes or ACI)
+    * During rendering, it examines the environment compute kind and delegates to the appropriate renderer.
+1. Concrete Renderer Implementations
+    * Platform-specific renderers like `aci.Renderer` and `container.Renderer` (Kubernetes) implement the same interface
+    * Resource-specific renderers like `gateway.Renderer` and `aci_gateway.Renderer` handle specific resource types
```

Comment:
I assume these need to be renamed as a part of this work? Or perhaps as a part of the "Radius core platform extensibility" work?

Reply:
We don't plan to rename these right now, but I'm open to discuss doing so.